aws_batch/indexing.py

```diff
@@ -25,20 +27,62 @@ def execute_command(command):
     output = subprocess.check_output(command, shell=True)
     return output
 
+def split_fasta(fasta_file, max_fasta_part_size):
+    fasta_file_list = []
+    part_idx = 0; current_size = 0
+    current_output_file_name = "%s.%d" % (fasta_file, part_idx)
+    current_output_file = open(current_output_file_name, 'wb')
+    fasta_file_list.append(current_output_file_name)
+    with open(fasta_file, 'rb') as input_f:
+        current_read = input_f.readline()
+        for line in input_f:
+            # Check if we have to switch different output fasta file
+            if current_size > max_fasta_part_size:
+                current_output_file.close()
+                part_idx += 1; current_size = 0
+                current_output_file_name = "%s.%d" % (fasta_file, part_idx)
+                current_output_file = open(current_output_file_name, 'wb')
+                fasta_file_list.append(current_output_file_name)
+
+            if line[0] == '>': # got a new read
+                current_output_file.write(current_read)
+                current_size += len(current_read)
+                current_read = line
+            else:
+                current_read += line
+        current_output_file.write(current_read)
+        current_output_file.close()
+    return fasta_file_list
```

Comment:
I see a couple of bugs here.

First, current_size is not initialized when it is first accessed.

Second, the last read in the input file is lost (never output).

Reply:
line 32: part_idx = 0; current_size = 0 
line 53: current_output_file.write(current_read)